File: tests/r27b1b/test_no_model_assets_committed.ts
import test from "node:test";
import assert from "node:assert/strict";
import { spawnSync } from "node:child_process";

test("no model assets or tokenizer artifacts are tracked", () => {
  const result = spawnSync("git", ["ls-files"], { encoding: "utf8" });
  assert.equal(result.status, 0);
  assert.doesNotMatch(result.stdout, /\.(pt|pth|safetensors|ckpt|onnx|bin|gguf)$/m);
  assert.doesNotMatch(result.stdout, /^(artifacts|web\/another_brain|web\/another_brain_chat|src\/browser_runtime).*tokenizer\.json$/m);
  assert.doesNotMatch(result.stdout, /^artifacts\/r27b1b\//m);
});

File: tests/r27b1c/test_no_model_assets_committed.py
import subprocess
import unittest
from pathlib import Path


ROOT = Path(__file__).resolve().parents[2]


class R27B1CNoModelAssetsCommittedTests(unittest.TestCase):
    def test_no_b_line_model_assets_or_artifacts_are_tracked(self):
        tracked = subprocess.run(["git", "ls-files"], cwd=ROOT, text=True, capture_output=True, check=True).stdout
        self.assertNotRegex(tracked, r"^(artifacts/(?!\.gitkeep$)|web/another_brain/.*\.(pt|pth|safetensors|ckpt|onnx|bin|gguf)$)")
        self.assertNotRegex(tracked, r"^(web/another_brain|web/another_brain_chat|src/browser_runtime).*tokenizer\.json$")
        self.assertNotIn("artifacts/r27b1c", tracked)


if __name__ == "__main__":
    unittest.main()
